src/java/fr/paris/lutece/plugins/forms/service/FormsMultiviewAuthorizationService.java

```diff
@@ -121,6 +123,12 @@ public boolean isUserAuthorizedOnFormResponse( HttpServletRequest request, int n
                 List<Integer> listIdFormResponse = listFormResponseItem.stream( ).map( FormResponseItem::getIdFormResponse ).collect( Collectors.toList( ) );
                 bIsUserAuthorizedOnFormResponse = listIdFormResponse.contains( nIdFormResponse );
             }
+            if(bIsUserAuthorizedOnFormResponse) {
+                User user = AdminUserService.getAdminUser(request);
+                FormResponse formResponse = FormResponseHome.findByPrimaryKey(nIdFormResponse);
+                Form form = FormHome.findByPrimaryKey(formResponse.getFormId());
```

Comment:
this will be executed for each response,  it should be optimized

Reply:
This function ( isUserAuthorizedOnFormResponse ) is only called once with the view view_form_response_details ( and 4 times in the tests). This is a view to see only on response at the time.  You pointed it out the case where this function is called in an loop with all the responses of a form, then it would be called for every responses. So I have made the check (form access within workgoup) in a separed function, by adding this commit :  
https://github.com/lutece-platform/lutece-form-plugin-forms/pull/335/commits/06516aefb1f298ea80775c48e3cd6e75f1c3cd78